### ISSUE TITLE
Concent still generates FileTransferTokens with base64-encoded public keys

### ISSUE THREAD

**Issue by cameel** [golemfactory/concent#283]
@shadeofblue reports a problem with uploading a file to the storage cluster on staging. Details below.

Before closing this issue please add a comment explaining what exactly was wrong.

### Request
``` bash
curl -X POST http://staging.concent.golem.network/upload/ \
    --header      "Authorization: Golem D6UAAAAAWtRjhwA/PHFb+hhQODlUe0dBkQ3GzGeReuPekSXbZUq814SFWTc0PaC8AtvSiesHBT66HU927HVoSwn/J1JaOR4Pm6bfAYaCanN1YnRhc2tfaWR4JDMwZTcwOTZiLWE2MTAtNzNiOC01ODQxLTcwZmY5MGZhNWY4ZoJ4GXRva2VuX2V4cGlyYXRpb25fZGVhZGxpbmUaWtXtR4J3c3RvcmFnZV9jbHVzdGVyX2FkZHJlc3N4JWh0dHA6Ly9zdGFnaW5nLmNvbmNlbnQuZ29sZW0ubmV0d29yay+CeBxhdXRob3JpemVkX2NsaWVudF9wdWJsaWNfa2V5WFhETkdaeWVTQmxLWXVJZVJabjRhTFM3aHl5Tkx2VVN3RVo5ZEpPVDJZNWhyV2h6b1NXbVhZN2EvM1lyU1YrM0IvVXpOV3dpUkw4R05FQm5UTmxBMDg2dz09gmlvcGVyYXRpb25mdXBsb2FkgmVmaWxlc4GjaGNoZWNrc3VteC1zaGExOjU1Zjg2YjFhN2E5Y2UzMTUzZGE2NTc3YmViNWIxMmUyMDY3MDNlOTFkcGF0aHiBYmxlbmRlci9yZXN1bHQvMWIwOTExYTgtODM1MC02NjQ1LWNhZmUtZTJjOWY5NjBlOGJhLzFiMDkxMWE4LTgzNTAtNjY0NS1jYWZlLWUyYzlmOTYwZThiYS4zMGU3MDk2Yi1hNjEwLTczYjgtNTg0MS03MGZmOTBmYTVmOGYuemlwZHNpemUaAAQAAA==" \
    --header      "Concent-Client-Public-Key: DNGZyeSBlKYuIeRZn4aLS7hyyNLvUSwEZ9dJOT2Y5hrWhzoSWmXY7a/3YrSV+3B/UzNWwiRL8GNEBnTNlA086w==" \
    --header      "Content-Type: application/octet-stream" \
    --header      "Concent-Upload-Path: blender/result/1b0911a8-8350-6645-cafe-e2c9f960e8ba/1b0911a8-8350-6645-cafe-e2c9f960e8ba.30e7096b-a610-73b8-5841-70ff90fa5f8f.zip" \
    --data-binary @/tmp/tmppy99hn3r
```

### Response
Status: HTTP 401

``` json
{"error_message": "Access Denied", "error_code": "unauthorized"}
```

**Comment by Jakub89** [comment]
FileTransferToken in this curl(golem scripts) looks like this:
```
FileTransferToken(header=MessageHeader(type_=4005, timestamp=1523954939, encrypted=False), sig=b'Ru\x9c\x89\x0c\x00\xe5\xb6\x1d\xbcS\xa3s3\x84\x02\xfc`F\xb2\x99\xe7;"\xa7\x07\x18\xdb\xdfz\xd0\xf5\x7f\xad\xfb\xf7\x92WG\x01\xe6Y\xef"\x92\xf8\xac\xad\xc8\xd7\x8c\x07\xb0\xda\xc5\x1e\xdc2.1\x1b\xe17\xf6\x01', 
slots=[['subtask_id', 'c8822b07-c139-fdf8-8f56-8deec8454828'], 
['token_expiration_deadline', 1524055738], ['storage_cluster_address', 'http://staging.concent.golem.network/'], 
['authorized_client_public_key', b'+xwU2H7BnLU/6nsPlqfU2bZZDcP8JFRzCI9ZrcOpsuUDzv45Dbsjs8fJ+U/T/aVz+fvjexYZsy/Kz7NK/u1J0w=='], 
['operation', 'upload'], ['files', [{'checksum': 'sha1:c2122c6aa40e749c5e9c0db6b84c61e18cd4b231', 'path': 'blender/result/1247165e-ce35-e8e3-a157-2e26aca65cad/1247165e-ce35-e8e3-a157-2e26aca65cad.c8822b07-c139-fdf8-8f56-8deec8454828.zip', 'size': 65536}]]])
```
`authorized_client_public_key` is in base64, but should be in `bytes`. Gatekeeper give "Access Denied" on this check:
```
if client_public_key != client_public_key_base64:
        return gatekeeper_access_denied_response('authorized_client_public_key is different then Concent-Client-Public-Key header...)
```
where `client_public_key = request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY']`(base64) and `client_public_key_base64 = (b64encode(loaded_golem_message.authorized_client_public_key)).decode('ascii')`

**Comment by cameel** [comment]
The issue is that @kbeker's fix for #259 (bc3da1f2ed404281efac5c52d5fb990df60828ad) was incomplete. It changed `authorized_client_public_key` in one place where we generate tokens for checking upload status but left it base64-encoded in the other place where we generate it for the client.

- Go over all our tests and make sure that the value in `authorized_client_public_key` field is never base64-encoded and is `bytes` (not `str`).
    - Note that the value being `bytes` on its own is not enough. It does not mean that the data is not base64-encoded.
- Fix the code
    - `authorized_client_public_key` should always be `bytes` and contain binary, unencoded data.
    - Make sure that base64-encoded keys are always using the `str` type, not `bytes` for consistency.
    - Make sure that non-encoded stuff never has `encoded_` prefix and vice versa.
    - In https://github.com/golemfactory/concent/blob/master/concent_api/gatekeeper/views.py#L148-L149: `client_public_key_base64` looks like base64-encoded `client_public_key` but it's not true. Call it `authorized_client_public_key_base64` to prevent mistakes in the future.
